R/plot_methods.R

```diff
@@ -0,0 +1,44 @@
+#' @importFrom ggplot2 autoplot .data
+#' @export
+ggplot2::autoplot
```

Comment:
Black magic! Never seen this pattern, but I'm OK with it.

Reply:
This is the general pattern to use when you create an S3 method for the object in your package but where the S3 generic lives in a different package. In this case `autoplot()` is the generic from `ggplot2` and we want to create a new method for that function.

So we need to import it for internal use, but also reexport it so that our users can call `autoplot(<summrt_summary>)` rather than `autoplot.summrt_summary(<summrt_summary>)`.

Reply:
Though maybe we could use `@exportS3method` alone instead? This has been updated recently:
https://r-pkgs.org/dependencies-in-practice.html#imports-and-exports-related-to-s3

My understanding is based off of, e.g., `{dplyr}`. For example, they make an `as_tibble.grouped_df()` [method](https://github.com/tidyverse/dplyr/blob/e4e9a295a373b85e02ae084a23f12e9212a72b98/R/grouped-df.R#L209) while the relevant `as_tibble()` generic is imported and rexported [here](https://github.com/tidyverse/dplyr/blob/e4e9a295a373b85e02ae084a23f12e9212a72b98/R/reexport-tibble.R#L27)

Reply:
If CRAN/R CMD check doesn't complain, then I'm good with it!